src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -7119,7 +7120,7 @@ def create_subnet(cmd, resource_group_name, virtual_network_name, subnet_name,
     if delegations:
         subnet.delegations = delegations
 
-    if disable_private_endpoint_network_policies is True:
+    if disable_private_endpoint_network_policies is None or disable_private_endpoint_network_policies is True:
```

Comment:
Why not change the function signature
```
disable_private_endpoint_network_policies=True,
```

Reply:
Did some research, the reason why default argument "True" becomes "1", it's a kind of distinguishing:
https://github.com/microsoft/knack/blob/b4d9e66b469ef7c18a1f228f803505b7f550bdc1/knack/commands.py#L122-L124